cookbooks/bcpc-hadoop/recipes/datanode.rb

```diff
@@ -33,6 +33,32 @@
   process_limit 65536
 end
 
+# need to ensure hdfs user is in hadoop and hdfs
+# groups. Packages will not add hdfs if it
+# is already created at install time (e.g. if
+# machine is using LDAP for users).
+# Similarly, yarn needs to be in the hadoop
+# group to run the LCE
+group 'hadoop' do
+  # use manage as if the group does not exist
+  # we do not want an exception
+  action :manage
+  members ['hdfs', 'yarn']
```

Comment:
I do not know why; I am unaware of any particular directory which is group `hadoop`?